src/figuresClasses.ts

```diff
@@ -2,10 +1,98 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
+  ) {
+    this.a = a;
+    this.b = b;
+    this.c = c;
+    this.color = color;
+    this.shape = 'triangle';
+
+    if (
+      this.a === 0 ||
+      this.b === 0 ||
+      this.c === 0 ||
+      this.a >= this.b + this.c ||
+      this.b >= this.a + this.c ||
+      this.c >= this.a + this.b
+    ) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
```

Comment:
According to the checklist, the `getArea` function should not accept any parameters and should use the object's fields. This is correctly implemented here.